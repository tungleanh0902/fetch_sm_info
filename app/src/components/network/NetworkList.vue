<template>
  <section>
    <h3>{{ sectionTitle }}</h3>
    <ul class="network-list">
      <li
        v-for="network in networks"
        :key="network.id"
        class="select-network-item"
        :data-network="network.id"
        @click="network.chain_id ? selectNetworkHandler(network) : false"
      >
        <NetworkItem :network-item="network" :disabled="disabled" />
      </li>
    </ul>
  </section>
</template>

<script>
import { mapGetters, mapState } from "vuex"
import NetworkItem from "./NetworkItem"

export default {
  name: `network-list`,
  components: {
    NetworkItem,
  },
  props: {
    networks: {
      type: Array,
      required: true,
    },
    disabled: {
      type: Boolean,
      required: false,
    },
    sectionTitle: {
      type: String,
      required: true,
    },
  },
  computed: {
    ...mapState(["session"]),
    ...mapGetters([{ networkId: `network` }, `isExtension`]),
    whichFlow() {
      if (this.$route.name === "select-network-recover") {
        return `/recover`
      } else if (this.$route.name === "select-network-create") {
        return `/create`
      } else {
        return ``
      }
    },
  },
  methods: {
    async selectNetworkHandler(network) {
      if (this.networkId !== network.id) {
        await this.$store.dispatch(`setNetwork`, network)
      }
      if (this.$route.name !== "networks") {
        this.$router.push(this.whichFlow)
        return
      }

      // enter the network
      if (
        this.session.allSessionAddresses.find(
          ({ networkId }) => networkId === network.id
        )
      ) {
        this.$router.push({
          name: `portfolio`,
          params: { networkId: network.slug },
        })
        // if no active session found then take to the validators table
      } else {
        this.$router.push({
          name: `validators`,
          params: { networkId: network.slug },
        })
      }
    },
  },
}
</script>
<style scoped>
h3 {
  margin: 0 0 0.25rem 1rem;
  color: var(--dim);
  font-size: var(--sm);
  font-weight: 500;
}

.network-list {
  margin-bottom: 4rem;
}
</style>
