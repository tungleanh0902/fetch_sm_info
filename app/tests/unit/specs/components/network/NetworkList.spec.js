import { shallowMount, createLocalVue } from "@vue/test-utils"
import NetworkList from "network/NetworkList"

const localVue = createLocalVue()

describe(`NetworkList`, () => {
  let wrapper, $store

  const networks = [
    {
      id: "gaia-testnet",
      chain_id: "gaia-123",
      logo_url: "cosmos-logo.png",
      testnet: true,
      title: "Cosmos Hub Test",
    },
    {
      id: "cosmoshub",
      chain_id: "cosmoshub",
      logo_url: "cosmos-logo.png",
      testnet: false,
      title: "Cosmos Hub",
    },
  ]

  beforeEach(() => {
    $store = {
      dispatch: jest.fn(),
      getters: {
        network: `cosmoshub`,
      },
    }
    wrapper = shallowMount(NetworkList, {
      localVue,
      propsData: {
        networks,
        sectionTitle: `section title`,
      },
      mocks: {
        $store,
        $route: {
          name: `some-random-route`,
        },
        $router: {
          push: jest.fn(),
        },
      },
      stubs: [`router-link`],
    })
  })

  it(`shows a list of networks`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
