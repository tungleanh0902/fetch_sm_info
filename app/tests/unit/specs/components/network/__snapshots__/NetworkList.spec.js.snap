// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkList shows a list of networks 1`] = `
<section>
  <h3>
    section title
  </h3>
   
  <ul
    class="network-list"
  >
    <li
      class="select-network-item"
      data-network="gaia-testnet"
    >
      <networkitem-stub
        networkitem="[object Object]"
      />
    </li>
    <li
      class="select-network-item"
      data-network="cosmoshub"
    >
      <networkitem-stub
        networkitem="[object Object]"
      />
    </li>
  </ul>
</section>
`;
